docs/usage-guide/topics/ch04-connection.md

```diff
@@ -1,6 +1,8 @@
 # TLS Connections
 
-Users will need to create a `s2n_connection` struct to store all of the state necessary for a TLS connection. Call `s2n_connection_new()` to create a new server or client connection. Call `s2n_connection_free()` to free the memory allocated for this struct when no longer needed.
+A TLS connection is a secure and encrypted channel established between two communicating peers. A TLS connection is used to send and receive data, perform TLS handshake negotiations, send alert messages, and etc. Read [TLS Connections](./ch04-connection.md) for information about connections, and [Sending and Receiving](./ch07-io.md) for io interactions. TLS connections are configured by TLS configs, which contains a collection of TLS settings. TLS configs apply rules on TLS connections by defining connection specs, such as supported certificate authorities. Read [Configuring the Connection](./ch05-config.md) for information about TLS connection and config interactions.
```

Comment:
You need to give introductions for s2n-tls connections, but a user of a TLS library can probably be assumed to know what a TLS connection is. It also looks like you're using "TLS connection" and "s2n-tls connection" interchangeably tho. And "TLS configs" aren't a standard part of TLS, those are an s2n-tls implementation detail.

And I also still don't think this paragraph will really be useful to a new user trying to understand the overall flow of the library. You give examples of possible things you can do with connections, but it's really not clear enough to be helpful. What do I need to do with a connection to use TLS? You also include a link to this section, which is odd.

You also go too deep into "TLS configs"-- configs have their own section, they should at most be referenced here.